{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pypelines.ts_forecast_pipeline as pipe\n",
    "from pypelines import utils\n",
    "import pandas as pd\n",
    "airline = pd.read_csv(\"pypelines/datasets/timeseries_forecast/airline.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Date</th>\n",
       "      <th>Passengers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1949-01</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1949-02</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1949-03</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1949-04</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1949-05</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>1960-08</td>\n",
       "      <td>606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>1960-09</td>\n",
       "      <td>508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>1960-10</td>\n",
       "      <td>461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>1960-11</td>\n",
       "      <td>390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>1960-12</td>\n",
       "      <td>432</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Date  Passengers\n",
       "0    1949-01         112\n",
       "1    1949-02         118\n",
       "2    1949-03         132\n",
       "3    1949-04         129\n",
       "4    1949-05         121\n",
       "..       ...         ...\n",
       "139  1960-08         606\n",
       "140  1960-09         508\n",
       "141  1960-10         461\n",
       "142  1960-11         390\n",
       "143  1960-12         432\n",
       "\n",
       "[144 rows x 2 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "airline\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "tsf = pipe.TSForecastPipeline(data = \"airline\",models=['Exponential Smoothing'],\n",
    "                 date_column = 'Date',\n",
    "                 date_format = \"%Y-%m\",\n",
    "                 frequency = \"M\",\n",
    "                 forecast_horizon = [1,2,3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "tsf.code_to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fitting 132 folds for each of 1 candidates, totalling 132 fits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=10.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=11.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=12.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=13.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=14.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=15.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=16.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=17.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=18.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=19.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=20.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=21.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=22.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=23.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=24.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=25.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=26.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=27.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=28.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=29.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=30.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=31.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=32.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=33.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=34.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=35.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=36.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=37.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=38.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=39.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=40.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=41.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=42.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=43.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=44.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=45.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=46.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=47.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=48.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=49.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=50.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=51.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=52.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=53.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=54.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=55.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=56.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=57.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=58.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=59.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=60.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=61.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=62.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=63.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=64.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=65.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=66.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=67.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=68.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=69.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=70.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=71.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=72.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=73.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=74.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=75.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=76.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=77.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=78.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=79.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=80.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=81.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=82.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=83.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=84.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=85.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=86.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=87.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=88.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=89.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=90.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=91.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=92.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=93.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=94.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=95.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=96.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=97.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=98.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=99.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=100.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=101.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=102.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=103.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=104.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=105.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=106.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=107.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=108.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=109.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=110.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=111.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=112.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=113.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=114.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=115.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=116.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=117.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=118.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=119.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=120.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=121.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=122.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=123.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=124.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=125.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=126.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=127.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=128.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=129.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=130.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=131.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=132.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=133.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=134.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=135.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=136.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=137.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=138.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=139.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=140.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n",
      "c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_evaluation\\_functions.py:413: FitFailedWarning: \n",
      "                In evaluate, fitting of forecaster ExponentialSmoothing failed,\n",
      "                you can set error_score='raise' in evaluate to see\n",
      "                the exception message. Fit failed for len(y_train)=141.\n",
      "                The score will be set to nan.\n",
      "                Failed forecaster with parameters: ExponentialSmoothing(sp=12).\n",
      "                \n",
      "  result = _evaluate_window(\n"
     ]
    },
    {
     "ename": "NotFittedError",
     "evalue": "All fits of forecaster failed,\n                set error_score='raise' to see the exceptions.\n                Failed forecaster: ExponentialSmoothing()",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNotFittedError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[12], line 44\u001b[0m\n\u001b[0;32m     37\u001b[0m ExponentialSmoothing_model \u001b[39m=\u001b[39m ExponentialSmoothing()\n\u001b[0;32m     39\u001b[0m ExponentialSmoothing_gscv \u001b[39m=\u001b[39m ForecastingGridSearchCV(\n\u001b[0;32m     40\u001b[0m     forecaster\u001b[39m=\u001b[39mExponentialSmoothing_model,\n\u001b[0;32m     41\u001b[0m     param_grid\u001b[39m=\u001b[39mExponentialSmoothing_param_grid,\n\u001b[0;32m     42\u001b[0m     cv\u001b[39m=\u001b[39mcv,\n\u001b[0;32m     43\u001b[0m     verbose\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m---> 44\u001b[0m ExponentialSmoothing_gscv\u001b[39m.\u001b[39;49mfit(x_train_preprocessed)\n\u001b[0;32m     46\u001b[0m \u001b[39m# get the prediction on the test data\u001b[39;00m\n\u001b[0;32m     47\u001b[0m ExponentialSmoothing_y_pred \u001b[39m=\u001b[39m ExponentialSmoothing_gscv\u001b[39m.\u001b[39mpredict(fh\u001b[39m=\u001b[39m[\u001b[39m1\u001b[39m, \u001b[39m2\u001b[39m, \u001b[39m3\u001b[39m])\n",
      "File \u001b[1;32mc:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\base\\_base.py:348\u001b[0m, in \u001b[0;36mBaseForecaster.fit\u001b[1;34m(self, y, X, fh)\u001b[0m\n\u001b[0;32m    346\u001b[0m \u001b[39m# we call the ordinary _fit if no looping/vectorization needed\u001b[39;00m\n\u001b[0;32m    347\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m vectorization_needed:\n\u001b[1;32m--> 348\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_fit(y\u001b[39m=\u001b[39;49my_inner, X\u001b[39m=\u001b[39;49mX_inner, fh\u001b[39m=\u001b[39;49mfh)\n\u001b[0;32m    349\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m    350\u001b[0m     \u001b[39m# otherwise we call the vectorized version of fit\u001b[39;00m\n\u001b[0;32m    351\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_vectorize(\u001b[39m\"\u001b[39m\u001b[39mfit\u001b[39m\u001b[39m\"\u001b[39m, y\u001b[39m=\u001b[39my_inner, X\u001b[39m=\u001b[39mX_inner, fh\u001b[39m=\u001b[39mfh)\n",
      "File \u001b[1;32mc:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\model_selection\\_tune.py:231\u001b[0m, in \u001b[0;36mBaseGridSearch._fit\u001b[1;34m(self, y, X, fh)\u001b[0m\n\u001b[0;32m    229\u001b[0m \u001b[39m# Raise error if all fits in evaluate failed because all score values are NaN.\u001b[39;00m\n\u001b[0;32m    230\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbest_index_ \u001b[39m==\u001b[39m \u001b[39m-\u001b[39m\u001b[39m1\u001b[39m:\n\u001b[1;32m--> 231\u001b[0m     \u001b[39mraise\u001b[39;00m NotFittedError(\n\u001b[0;32m    232\u001b[0m         \u001b[39mf\u001b[39m\u001b[39m\"\"\"\u001b[39m\u001b[39mAll fits of forecaster failed,\u001b[39m\n\u001b[0;32m    233\u001b[0m \u001b[39m        set error_score=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mraise\u001b[39m\u001b[39m'\u001b[39m\u001b[39m to see the exceptions.\u001b[39m\n\u001b[0;32m    234\u001b[0m \u001b[39m        Failed forecaster: \u001b[39m\u001b[39m{\u001b[39;00m\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mforecaster\u001b[39m}\u001b[39;00m\u001b[39m\"\"\"\u001b[39m\n\u001b[0;32m    235\u001b[0m     )\n\u001b[0;32m    236\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbest_score_ \u001b[39m=\u001b[39m results\u001b[39m.\u001b[39mloc[\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbest_index_, \u001b[39mf\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mmean_\u001b[39m\u001b[39m{\u001b[39;00mscoring_name\u001b[39m}\u001b[39;00m\u001b[39m\"\u001b[39m]\n\u001b[0;32m    237\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbest_params_ \u001b[39m=\u001b[39m results\u001b[39m.\u001b[39mloc[\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbest_index_, \u001b[39m\"\u001b[39m\u001b[39mparams\u001b[39m\u001b[39m\"\u001b[39m]\n",
      "\u001b[1;31mNotFittedError\u001b[0m: All fits of forecaster failed,\n                set error_score='raise' to see the exceptions.\n                Failed forecaster: ExponentialSmoothing()"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "from sktime import *\n",
    "from sklearn.metrics import mean_squared_error\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "date_col = 'Date'\n",
    "features = list(airline.columns)\n",
    "feature_df = airline[features]\n",
    "\n",
    "x_train = airline[features]\n",
    "x_train_preprocessed = x_train\n",
    "x_train_preprocessed[date_col] = pd.to_datetime(x_train_preprocessed[date_col],format = \"%Y-%m\")\n",
    "x_train_preprocessed = x_train_preprocessed.set_index(date_col)\n",
    "x_train_preprocessed.index = x_train_preprocessed.index.to_period(freq = \"M\")\n",
    "\n",
    "model_comparison_list = []\n",
    "\n",
    "##### End of Data Processing Pipeline #####\n",
    "\n",
    "\n",
    "##### Model Pipeline for Exponential Smoothing #####\n",
    "\n",
    "from sktime.forecasting.exp_smoothing import ExponentialSmoothing\n",
    "from sktime.forecasting.model_selection import ExpandingWindowSplitter,ForecastingGridSearchCV,ExpandingWindowSplitter\n",
    "from sklearn.metrics import mean_squared_error,make_scorer,r2_score,explained_variance_score\n",
    "import matplotlib.pyplot as plt\n",
    "cv = ExpandingWindowSplitter(fh=[1, 2, 3])\n",
    "\n",
    "ExponentialSmoothing_param_grid = {\n",
    "\"sp\": np.arange(12, 13, 1),\n",
    "}\n",
    "\n",
    "\n",
    "ExponentialSmoothing_model = ExponentialSmoothing()\n",
    "\n",
    "ExponentialSmoothing_gscv = ForecastingGridSearchCV(\n",
    "    forecaster=ExponentialSmoothing_model,\n",
    "    param_grid=ExponentialSmoothing_param_grid,\n",
    "    cv=cv,\n",
    "    verbose=1)\n",
    "ExponentialSmoothing_gscv.fit(x_train_preprocessed)\n",
    "\n",
    "# get the prediction on the test data\n",
    "ExponentialSmoothing_y_pred = ExponentialSmoothing_gscv.predict(fh=[1, 2, 3])\n",
    "##### End of Model Pipeline for Exponential Smoothing #####\n",
    "##### Model Comparison #####\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "ExponentialSmoothing_param_grid = {\n",
    "\"sp\": np.arange(12, 13, 1),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'sp': array([12])}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ExponentialSmoothing_param_grid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unexpected exception formatting exception. Falling back to standard exception\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"__init__.pxd\", line 942, in numpy.import_array\n",
      "RuntimeError: module compiled against API version 0x10 but this version of numpy is 0xf . Check the section C-API incompatibility at the Troubleshooting ImportError section at https://numpy.org/devdocs/user/troubleshooting-importerror.html#c-api-incompatibility for indications on how to solve this problem .\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3460, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Local\\Temp\\ipykernel_597156\\3746879201.py\", line 7, in <module>\n",
      "    forecaster.fit(y)\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\base\\_base.py\", line 348, in fit\n",
      "    self._fit(y=y_inner, X=X_inner, fh=fh)\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\base\\adapters\\_statsmodels.py\", line 55, in _fit\n",
      "    self._fit_forecaster(y, X)\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\sktime\\forecasting\\exp_smoothing.py\", line 170, in _fit_forecaster\n",
      "    from statsmodels.tsa.holtwinters import (\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\statsmodels\\tsa\\holtwinters\\__init__.py\", line 1, in <module>\n",
      "    from statsmodels.tsa.holtwinters.model import (\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\statsmodels\\tsa\\holtwinters\\model.py\", line 35, in <module>\n",
      "    from statsmodels.tsa.exponential_smoothing.ets import (\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\statsmodels\\tsa\\exponential_smoothing\\ets.py\", line 166, in <module>\n",
      "    from statsmodels.tsa.exponential_smoothing import base\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\statsmodels\\tsa\\exponential_smoothing\\base.py\", line 21, in <module>\n",
      "    from statsmodels.tsa.statespace.tools import _safe_cond\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\statsmodels\\tsa\\statespace\\tools.py\", line 14, in <module>\n",
      "    from . import (_initialization, _representation, _kalman_filter,\n",
      "  File \"statsmodels\\tsa\\statespace\\_initialization.pyx\", line 1, in init statsmodels.tsa.statespace._initialization\n",
      "  File \"statsmodels\\tsa\\statespace\\_representation.pyx\", line 1, in init statsmodels.tsa.statespace._representation\n",
      "  File \"statsmodels\\tsa\\statespace\\_kalman_filter.pyx\", line 1, in init statsmodels.tsa.statespace._kalman_filter\n",
      "  File \"statsmodels\\tsa\\statespace\\_kalman_smoother.pyx\", line 41, in init statsmodels.tsa.statespace._kalman_smoother\n",
      "  File \"__init__.pxd\", line 944, in numpy.import_array\n",
      "ImportError: numpy.core.multiarray failed to import\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    stb = self.InteractiveTB.structured_traceback(\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1288, in structured_traceback\n",
      "    return FormattedTB.structured_traceback(\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1177, in structured_traceback\n",
      "    return VerboseTB.structured_traceback(\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1049, in structured_traceback\n",
      "    formatted_exceptions += self.format_exception_as_a_whole(etype, evalue, etb, lines_of_context,\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 935, in format_exception_as_a_whole\n",
      "    self.get_records(etb, number_of_lines_of_context, tb_offset) if etb else []\n",
      "  File \"C:\\Users\\Kreshnaa Raam\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1003, in get_records\n",
      "    lines, first = inspect.getsourcelines(etb.tb_frame)\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\inspect.py\", line 1121, in getsourcelines\n",
      "    lines, lnum = findsource(object)\n",
      "  File \"c:\\Users\\Kreshnaa Raam\\AppData\\Local\\Programs\\Python\\Python310\\lib\\inspect.py\", line 958, in findsource\n",
      "    raise OSError('could not get source code')\n",
      "OSError: could not get source code\n"
     ]
    }
   ],
   "source": [
    "from sktime.datasets import load_airline\n",
    "from sktime.forecasting.exp_smoothing import ExponentialSmoothing\n",
    "y = load_airline()\n",
    "forecaster = ExponentialSmoothing(\n",
    "    trend='add', seasonal='multiplicative', sp=12\n",
    ")  \n",
    "forecaster.fit(y)  \n",
    "\n",
    "y_pred = forecaster.predict(fh=[1,2,3]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
